src/OpenFeature.Contrib.Providers.Flagd/FlagdProvider.cs

```diff
@@ -131,9 +107,24 @@ public static string GetProviderName()
         public override Metadata GetMetadata() => _providerMetadata;
 
         /// <summary>
-        ///     Return the Grpc client of the provider
+        ///     Return the resolver of the provider
         /// </summary>
-        public Service.ServiceClient GetClient() => _client;
+        internal Resolver.Resolver GetResolver() => _resolver;
+
+        /// <inheritdoc/>
+        public override Task Initialize(EvaluationContext context)
+        {
+            return Task.Run(() =>
+            {
+                this._resolver.Init();
+                this._status = ProviderStatus.Ready;
+
+            }).ContinueWith((t) =>
+            {
+                this._status = ProviderStatus.Error;
+                if (t.IsFaulted) throw t.Exception;
+            });
+        }
```

Comment:
@bacherfl @Kavindu-Dodan I added an initialize function, the old provider was built before this was implemented; it's no longer done in the contstructor.